service/app/controllers/metrics/CourseMetricsController.java

```diff
@@ -32,10 +34,17 @@
   }
 
   public Promise<Result> courseProgressV2(String batchId) {
-    final String limit = request().getQueryString(JsonKey.LIMIT);
-    final String offset = request().getQueryString(JsonKey.OFFSET);
-    final String sortBy = request().getQueryString(JsonKey.SORT_BY);
-
+    final String limit =
+        StringUtils.isEmpty(request().getQueryString(JsonKey.LIMIT))
```

Comment:
Read from query string only once instead of twice.

String limit = request().getQueryString(JsonKey.LIMIT);
limit = StringUtils.isEmpty(limit) ? "200" : limit;

Use constant DEFAULT_LIMIT.